lib/iex/lib/iex/introspection.ex

```diff
@@ -295,33 +295,57 @@ defmodule IEx.Introspection do
   end
 
   defp h_mod_fun(mod, fun) when is_atom(mod) do
-    if docs = Code.get_docs(mod, :docs) do
-      result =
-        for {{^fun, arity}, _, _, _, _} = doc <- docs, has_content?(doc) do
-          h_mod_fun_arity(mod, fun, arity)
+    cond do
+      docs = Code.get_docs(mod, :docs) ->
+        result =
+          for {{^fun, arity}, _, _, _, _} = doc <- docs, has_content?(doc) do
+            h_mod_fun_arity(mod, fun, arity)
+          end
+
+        cond do
+          result != [] ->
+            :ok
+
+          has_callback?(mod, fun) ->
+            :behaviour_found
+
+          true ->
+            :not_found
         end
 
-      cond do
-        result != [] ->
-          :ok
+      :code.is_loaded(mod) ->
+        result =
+          for {^fun, arity} <- mod.module_info(:exports) do
+            h_mod_fun_arity(mod, fun, arity)
+          end
 
-        has_callback?(mod, fun) ->
-          :behaviour_found
+        case result do
+          [] -> :not_found
+          [:not_found | _] -> :not_found
+          _ -> :ok
+        end
 
-        true ->
-          :not_found
-      end
-    else
-      :no_docs
+      true ->
+        :no_docs
     end
   end
 
   defp h_mod_fun_arity(mod, fun, arity) when is_atom(mod) do
     docs = Code.get_docs(mod, :docs)
+    spec = get_spec(mod, fun, arity)
 
     cond do
-      is_nil(docs) ->
-        :no_docs
+      is_nil(docs) and spec == [] ->
+        :not_found
+
+      is_nil(docs) and spec ->
+        print_doc(
+          "#{inspect(mod)}.#{fun}/#{arity}",
+          spec,
+          "This is an Erlang function and it currently does not have any Elixir-style docs"
```

Comment:
Actually, we can reach this clause if the module is compiled without docs. So I would do this:

```elixir
message =
  if function_exported?(mod, :__info__, 1) do
    "Module was compiled without docs. Showing only specs."
  else
    "Documentation is not available for non-Elixir modules. Showing only specs."
  end
```

Reply:
Ok cool. I've made the suggested changes but I'm having a bit of difficulty testing the scenario where the module is compiled without docs.
I've set up this sample file in the test case:
```
defmodule Sample do
  @moduledoc false

  @doc "foodoc"
  @spec foo() :: any()
  def foo(), do: 0

  def bar(), do: 0
end
```
but the call to `Code.get_docs(Sample, :docs)` always returns a non-empty List containing `[{{:bar, 0}, 8, :def, [], nil}, {{:foo, 0}, 4, :def, [], "foodoc"}]`; even if I remove the `@doc "foodoc"` I still don't get an empty List. Is this expected behaviour? From reading the doco at https://hexdocs.pm/elixir/writing-documentation.html#hiding-internal-modules-and-functions, I had thought that using `@moduledoc false` would actually give me either an empty list, or tuples with something like `{{:bar, 0}, 8, :def, [], false}`.

How would I go about compiling a sample file without docs?

Reply:
Ah, after playing with it a bit more it seems that I need to put `@doc false` before the function for `Code.get_docs/2` to return something like `[{{:bar, 0}, 8, :def, [], false}]`, only using `@moduledoc` is not enough.

So under what circumstances would we get an empty list from the `Code.get_docs/2` call?

Reply:
Whoops, I'm not supposed to get an empty list, it's supposed to be `nil`.

Reply:
Good call. You can still do this though: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/test/elixir/kernel/docs_test.exs#L29

Reply:
Cool!